src/common/util/serialization.js

```diff
@@ -0,0 +1,180 @@
+/*globals define*/
+/**
+ * @author pmeijer / https://github.com/pmeijer
+ */
+
+
+define([
+    'q',
+    'blob/util',
+    'common/util/util',
+    'common/storage/util',
+    'common/core/coreQ'
+], function (Q, blobUtil, commonUtils, storageUtils, Core) {
+
+    /**
+     * Functions for serializing a webgme project or models. These will work both on the client and server. <br>
+     * To include in your module (e.g. Plugin) require via 'common/util/serialization'.
+     * <br>
+     * <bold>Important!</bold> In order to export, the state must have been persisted in the database before exporting.
+     * If you make changes with the core make sure to persist and make a commit (it could be a headless commit too)
+     * before attempting to export.
+     * @module serialization
+     */
+    var exports = {};
+
+    /**
+     * Exports a snap-shot of the entire project-tree and uploads a webgmex-file on the blob storage.
+     * @param {ProjectInterface} project
+     * @param {BlobClient} blobClient
+     * @param {object} parameters - One of rootHash, commitHash and branchName and tagName must be given.
```

Comment:
we should specify the order of those parameters (so in case multiple would be given, there will be no confusion)